### PR TITLE
replaced webmozart/path-util with symfony/filesystem

### PR DESCRIPTION
**Pull request by a4blue**
The deprecation notice annoyed me, so i tried to change it :)
As far as i can tell, the change shouldn't be too drastic

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,10 +18,10 @@
     "symfony/config": "^3.4 || ^4.0 || ^5.0 || ^6.0",
     "symfony/console": "^3.4 || ^4.0 || ^5.0 || ^6.0",
     "symfony/dependency-injection": "^3.4 || ^4.0 || ^5.0 || ^6.0",
+    "symfony/filesystem": "^5.0",
```
Comment (jdreesen) [inline composer.json:21]: I think this should be `^5.4` as the `Path` class was added very recently: https://symfony.com/blog/new-in-symfony-5-4-filesystem-path-class

```suggestion
    "symfony/filesystem": "^5.4",
```

Comment (a4blue) [reply composer.json]: Thanks for your Comment, i added your suggestion :)

**Comment by DaveLiddament** [comment]
Thanks for this PR. I'm onboard with the aim of removing the deprecation notice. 

Given SARB's use case it will probably be used in with older code bases. I'm trying to keep SARB's dependencies as wide as possible. It's main dependencies are the Symfony components and my aim is for SARB to work with all supported versions of Symfony components (and PHP versions). The restriction to Symfony filesystem 5.4 is too tight. For adding a new Symfony library it would ideally satisfy the version constraints `4.4 || ^5.0 || ^6.0`.

To progress to your aim I've created the following issues:

- [ ] #108  Given path-utils stability; the latest version has had no code changes for 6 years and it's MIT licence, I will copy over the relevant code and tests to SARB
- [ ] #109  

I hope to spend some time over the xmas break to get this and a few of the other outstanding issues sorted. 

Again, thanks very much for this PR. I'm sorry I'm not accepting it at the moment. I hope the reasons why make sense, and the plan to meet your ultimate aim is acceptable.

**Comment by a4blue** [comment]
@DaveLiddament could you specify, which PHP Version you aim to Support ? As per composer.json i think PHP 7.3-8.1 is your target.
I also tried to fork the path-utils Project now, so that Projects that need replacement dont diverge too much apart :)
https://github.com/a4blue/path-util
I hope the fork is acceptable for you to use

**Comment by DaveLiddament** [comment]
Fixed by #110

**Pull request closed without merging**